tests/helpers/server.ts

```diff
@@ -0,0 +1,67 @@
+import { Locator, Page } from "@playwright/test";
+import { TIMEOUT } from "./constants";
+
+export type ServerSettingType = "checkbox" | "text" | "number" | "password";
+
+export const visitServerSettings = async (page: Page) => {
+  await page.goto("/ui/");
+  await page.getByRole("link", { name: "Settings", exact: true }).click();
+};
+
+export const activateSetting = async (page: Page, settingName: string) => {
+  await page
+    .locator("css=tr", { hasText: settingName })
+    .getByRole("button")
+    .click();
+};
+
+export const updateCheckbox = async (settingRow: Locator) => {
+  const checkbox = settingRow.locator(".p-checkbox__label");
+  await checkbox.click();
+};
+
+export const checkIfChangeButtonExist = async (settingRow: Locator) => {
+  const changeButton = settingRow.getByRole("button", {
+    name: "Change",
+    exact: true,
+  });
+  return changeButton.isVisible({ timeout: 1000 });
+};
+
+export const updateSetting = async (
+  page: Page,
+  settingName: string,
+  settingType: ServerSettingType,
+  content?: string,
+) => {
+  const settingRow = page.locator("css=tr", { hasText: settingName });
+  if (settingType === "checkbox") {
+    await updateCheckbox(settingRow);
+  } else {
+    const settingInput = settingRow.locator(`css=input[type=${settingType}]`);
+    if (content) {
+      await settingInput.fill(content);
+    }
+  }
+  await settingRow.getByRole("button", { name: "Save", exact: true }).click();
+  await page.waitForSelector("text=Setting updated.", TIMEOUT);
+};
+
+export const resetSetting = async (page: Page, settingName: string) => {
+  const settingRow = page.locator("css=tr", { hasText: settingName });
+  if (await checkIfChangeButtonExist(settingRow)) {
```

Comment:
Maybe this branch for the password type should be its own function `removePassword`?

Reply:
yeah that's better, will do that